contracts/oracle/PCVOracle.sol

```diff
@@ -0,0 +1,396 @@
+        return (1e18 * _lastLiquidBalance) / totalBalance;
+    }
+
+    /// @notice check if a venue is in the list of illiquid venues
+    /// @param illiquidVenue address to check
+    /// @return boolean whether or not the illiquidVenue is in the illiquid venue list
+    function isIlliquidVenue(
+        address illiquidVenue
+    ) external view returns (bool) {
+        return illiquidVenues.contains(illiquidVenue);
+    }
+
+    /// @notice check if a venue is in the list of illiquid venues
+    /// @param liquidVenue address to check
+    /// @return boolean whether or not the liquidVenue is in the illiquid venue list
+    function isLiquidVenue(address liquidVenue) external view returns (bool) {
+        return liquidVenues.contains(liquidVenue);
+    }
+
+    /// @notice check if a venue is in the list of liquid or illiquid venues
+    /// @param venue address to check
+    /// @return boolean whether or not the venue is part of the liquid or illiquid venue list
+    function isVenue(address venue) external view returns (bool) {
+        return liquidVenues.contains(venue) || illiquidVenues.contains(venue);
+    }
+
+    /// @notice get the total PCV balance by looping through the liquid and illiquid pcv deposits
+    /// @dev this function is meant to be used offchain, as it is pretty gas expensive.
+    function getTotalPcv()
+        external
+        view
+        returns (uint256 liquidPcv, uint256 illiquidPcv, uint256 totalPcv)
+    {
+        uint256 liquidVenueLength = liquidVenues.length();
+        uint256 illiquidVenueLength = illiquidVenues.length();
+
+        /// there will never be more than 100 total venues
+        /// so keep the math unchecked to save on gas
+        unchecked {
+            for (uint256 i = 0; i < liquidVenueLength; i++) {
+                address depositAddress = liquidVenues.at(i);
+                (Decimal.D256 memory oracleValue, bool oracleValid) = IOracle(
+                    venueToOracle[depositAddress]
+                ).read();
+                require(oracleValid, "PCVO: invalid oracle value");
+
+                liquidPcv +=
+                    (oracleValue.mul(1e18).asUint256() *
+                        IPCVDepositV2(depositAddress).balance()) /
+                    1e18;
+            }
+
+            for (uint256 i = 0; i < illiquidVenueLength; i++) {
+                address depositAddress = illiquidVenues.at(i);
+                (Decimal.D256 memory oracleValue, bool oracleValid) = IOracle(
+                    venueToOracle[depositAddress]
+                ).read();
+                require(oracleValid, "PCVO: invalid oracle value");
+
+                illiquidPcv +=
+                    (oracleValue.mul(1e18).asUint256() *
+                        IPCVDepositV2(depositAddress).balance()) /
+                    1e18;
+            }
+
+            totalPcv = liquidPcv + illiquidPcv;
+        }
+    }
+
```

Comment:
maybe we add a permissionless API that allows us to call accrue on all PCV deposits, both liquid and illiquid to get the system up to date

Reply:
and then call into the DynamicVoltSystemOracle after with the new liquid to illiquid ratio to compound interest and change the rate if needed.

Reply:
accrue() in the PCVdeposits will hook into the PCVOracle already, it would be a bit messy to add this feature IMO because we'd have to add a switch to disable notifications to DynamicVoltSystemOracle to prevent spamming it multiple times during each accrue(); and it also makes the PCVOracle a "system entry"